src/main/java/de/flapdoodle/embed/process/runtime/Starter.java

```diff
@@ -46,9 +46,15 @@ protected Starter(IRuntimeConfig config) {
 	}
 
 	public EXECUTABLE prepare(CONFIG config) {
-		
-		Distribution distribution = Distribution.detectFor(config.version());
-		
+		return prepare(config, null);
+	}
+
+	public EXECUTABLE prepare(CONFIG config, Distribution distribution) {
+
+		if (distribution == null) {
```

Comment:
You can remove this, if you change the other methods.. feel free to add a null-check:)